2019/d03p1LineCrossover.d

```diff
@@ -95,14 +100,33 @@ Pt linesCrossAt(Line l1, Line l2) {
   return Pt(0, 0);
 }
 
+struct Crossover {
+  Pt pt;
+  int steps;
+}
+
 auto findCrossovers(Line[][] wires) {
-  Pt[] crossovers;
+  Crossover[] crossovers;
+  auto l1steps = 0;
 
   foreach(l1; wires[0]) {
+    auto steps = l1steps;
+
     foreach(l2; wires[1]) {
       Pt crossover = linesCrossAt(l1, l2);
-      if (crossover != Pt.init) crossovers ~= crossover;
+      if (crossover != Pt.init) {
+        import std.math;
+        crossovers ~= Crossover(crossover,
+          steps + abs(crossover.x - l1.orig.x)
```

Comment:
I checked what Compiler Explorer had to say here and [LDC 1.80.0](https://godbolt.org/z/q4WRc6) does a good job of optimising this.